internal/attestation/vsa.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package attestation
+
+import (
+	"strings"
+
+	"github.com/in-toto/in-toto-golang/in_toto"
+
+	"github.com/enterprise-contract/ec-cli/internal/evaluator"
+	"github.com/enterprise-contract/ec-cli/internal/policy"
+	"github.com/enterprise-contract/ec-cli/internal/policy/source"
+)
+
+const (
+	// Make it visible elsewhere
+	PredicateVSAProvenance = "https://enterprisecontract.dev//verification_summary/v1"
+	StatmentVSA            = "https://in-toto.io/Statement/v1"
+)
+
+type ProvenanceStatementVSA struct {
+	in_toto.StatementHeader
+	Predicate predicate `json:"predicate"`
+}
+
+type policySource struct {
+	uri string
+}
+
+type attestationSource struct {
+	version string
+	digest  map[string]string
+}
+
+type predicate struct {
+	Verifier            map[string]string   `json:"verifier,omitempty"`
+	TimeVerified        string              `json:"timeVerified,omitempty"`
+	Policies            []policySource      `json:"policies,omitempty"`
+	InputAttestations   []attestationSource `json:"intputAttestations,omitempty"`
+	VerificationResult  string              `json:"verificationResult,omitempty"`
+	VerifiedRules       []string            `json:"verifiedRules,omitempty"`
+	VerifiedCollections []string            `json:"verfiedCollection,omitempty"`
+}
+
+func VsaFromImageValidation(verifiedTime string, results []evaluator.Outcome, policies []source.PolicySource, policy policy.Policy, attestations []Attestation) ProvenanceStatementVSA {
```

Comment:
For me it makes more sense for this to be in `internal/appicationsnapshot` and operated on Report there.

Having both policy.Policy and source.PolicySource in the function signature seems redundant

Reply:
yes, makes much more sense operating on report. Just moved it. Thanks.